functions/index.js

```diff
@@ -0,0 +1,128 @@
+'use strict';
+
+const functions = require('firebase-functions');
+const gcs = require('@google-cloud/storage')();
+const admin = require('firebase-admin');
+const jwt = require('jsonwebtoken');
+const fs = require('fs');
+
+admin.initializeApp(functions.config().firebase);
+
+const dataTypes = ['filenames', 'commit', 'result', 'sha', 'travis'];
+const repoSlug = functions.config().repo.slug;
+const secret = functions.config().secret.key;
+const bucket = gcs.bucket(functions.config().firebase.storageBucket);
+
+/** Copy valid data from /temp/screenshot/reports/$prNumber/$secureToken/ to /screenshot/reports/$prNumber */
+exports.copyData = functions.database.ref('/temp/screenshot/reports/{prNumber}/{token1}/{token2}/{token3}/{dataType}')
```

Comment:
Also, rather than using `temp/` is there something more descriptive we could use?